test/test.googleauth.js

```diff
@@ -129,6 +129,89 @@ describe('GoogleAuth', function() {
       });
     });
 
+    describe('.fromAPIKey', function () {
+      var API_KEY = 'test-123';
+      var STUB_PROJECT = 'my-awesome-project';
+      describe('Exception behaviour', function () {
+        var auth;
+        before(function () {
+          auth = new GoogleAuth();
+        });
+        it('Should error given an invalid api key', function (done) {
+          auth.fromAPIKey(null, function (err) {
+            assert(err instanceof Error);
+            done();
+          });
+        });
+      });
+      describe('Request/response lifecycle mocking', function () {
+        var ENDPOINT = '/events:report';
+        var RESPONSE_BODY = 'RESPONSE_BODY';
+        var BASE_URL = [
+          'https://clouderrorreporting.googleapis.com/v1beta1/projects',
+          STUB_PROJECT
+        ].join('/');
+        var auth;
+        beforeEach(function () {
+          auth = new GoogleAuth();
+          insertEnvironmentVariableIntoAuth(auth, 'GCLOUD_PROJECT', STUB_PROJECT);
+        });
+        afterEach(function () {
+          nock.cleanAll();
+        });
+        describe('With no added query string parameters', function () {
+          it('should make a request with the api key', function (done) {
+            var fakeService = nock(BASE_URL).post(ENDPOINT+'?key='+API_KEY)
```

Comment:
Nit: Prefer nock's `.query()` instead of putting the query parameters in the URL. https://github.com/node-nock/nock#specifying-request-query-string